tests/src/types/AtomicFixedPoint.cpp

```diff
@@ -0,0 +1,49 @@
+#include "openvic-simulation/types/fixed_point/Atomic.hpp"
+#include "openvic-simulation/types/fixed_point/FixedPoint.hpp"
+
+#include "Helper.hpp" // IWYU pragma: keep
+#include "Numeric.hpp" // IWYU pragma: keep
+#include <snitch/snitch_macros_check.hpp>
+#include <snitch/snitch_macros_misc.hpp>
+#include <snitch/snitch_macros_test_case.hpp>
+
+using namespace OpenVic;
+
+TEST_CASE("atomic_fixed_point_t Constructor methods", "[atomic_fixed_point_t][atomic_fixed_point_t-constructor]") {
+	CHECK(atomic_fixed_point_t{} == fixed_point_t::_0());
+	CHECK(atomic_fixed_point_t{12345l} == fixed_point_t::parse_raw(12345l));
+	CHECK(atomic_fixed_point_t{fixed_point_t::_1()} == fixed_point_t::_1());
+}
+
+TEST_CASE("atomic_fixed_point_t Operators", "[atomic_fixed_point_t][atomic_fixed_point_t-operators]") {
+	atomic_fixed_point_t a { fixed_point_t::_0() };
+
+	CHECK(a++ == fixed_point_t::_0());
+	CHECK(a == fixed_point_t::_1());
+
+	CHECK(++a == fixed_point_t::_2());
+
+	CHECK(a-- == fixed_point_t::_2());
+	CHECK(a == fixed_point_t::_1());
+
+	CHECK(--a == fixed_point_t::_0());
+
+	CHECK((a = fixed_point_t::_10()) == fixed_point_t::_10());
+	CHECK(a == fixed_point_t::_10());
+}
+
+TEST_CASE("atomic_fixed_point_t Operators", "[atomic_fixed_point_t][atomic_fixed_point_t-atomic-behavior]") {
```

Comment:
```suggestion
TEST_CASE("atomic_fixed_point_t Operators", "[atomic_fixed_point_t][.atomic_fixed_point_t-atomic-behavior]") {
```

If the test is too slow the dot hides it from normal testing so its only able to be explicitly executed.